tests/unit/cartography/intel/aws/ec2/test_launch_templates.py

```diff
@@ -97,127 +103,167 @@ def test_get_launch_template_versions_by_template_success():
+    # Transformed data needs Id, CreateTime as string timestamp, and flattened LaunchTemplateData fields
+    versions_transformed = [{
+        'LaunchTemplateId': 'lt-1',
+        'VersionNumber': 1,
+        'Id': 'lt-1-1',
+        'CreateTime': MOCK_CREATE_TIME_STR,
+    }]
+
+    mock_get_templates.return_value = templates_raw
+    mock_get_versions.return_value = versions_raw
+    mock_transform_templates.return_value = templates_transformed
+    mock_transform_versions.return_value = versions_transformed
 
     # Act
-    result_versions, result_templates = get_launch_template_versions(mock_session, region, templates)
+    sync_ec2_launch_templates(
+        mock_neo4j_session, mock_boto3_session, regions, FAKE_AWS_ACCOUNT_ID,
+        FAKE_UPDATE_TAG, COMMON_JOB_PARAMS,
+    )
 
     # Assert
-    assert result_versions == versions_found + versions_also_found
-    assert result_templates == [
-        {'LaunchTemplateId': 'lt-found'},
-        {'LaunchTemplateId': 'lt-also-found'},
-    ]
-    assert mock_get_by_template.call_count == 3
-    mock_get_by_template.assert_any_call(mock_session, 'lt-found', region)
-    mock_get_by_template.assert_any_call(mock_session, 'lt-not-found', region)
-    mock_get_by_template.assert_any_call(mock_session, 'lt-also-found', region)
-    mock_logger.warning.assert_called_once_with(
-        "Launch template %s no longer exists in region %s, skipping.",
-        'lt-not-found', region,
+    mock_get_templates.assert_called_once_with(mock_boto3_session, FAKE_REGION)
+    mock_get_versions.assert_called_once_with(mock_boto3_session, FAKE_REGION, templates_raw)
+    mock_transform_templates.assert_called_once_with(found_templates_filtered)
+    mock_load_templates.assert_called_once_with(
+        mock_neo4j_session, templates_transformed, FAKE_REGION, FAKE_AWS_ACCOUNT_ID, FAKE_UPDATE_TAG,
+    )
+    mock_transform_versions.assert_called_once_with(versions_raw)
+    mock_load_versions.assert_called_once_with(
+        mock_neo4j_session, versions_transformed, FAKE_REGION, FAKE_AWS_ACCOUNT_ID, FAKE_UPDATE_TAG,
+    )
+    mock_cleanup.assert_called_once_with(mock_neo4j_session, COMMON_JOB_PARAMS)
+    mock_logger.info.assert_called_once_with(
+        f"Syncing launch templates for region '{FAKE_REGION}' in account '{FAKE_AWS_ACCOUNT_ID}'.",
     )
 
 
 @patch('cartography.intel.aws.ec2.launch_templates.logger')
-@patch('cartography.intel.aws.ec2.launch_templates.get_launch_template_versions_by_template')
-def test_get_launch_template_versions_none_found(mock_get_by_template, mock_logger):
+@patch('cartography.intel.aws.ec2.launch_templates.cleanup')
+@patch('cartography.intel.aws.ec2.launch_templates.load_launch_template_versions')
+@patch('cartography.intel.aws.ec2.launch_templates.transform_launch_template_versions')
+@patch('cartography.intel.aws.ec2.launch_templates.load_launch_templates')
+@patch('cartography.intel.aws.ec2.launch_templates.transform_launch_templates')
```

Comment:
Rather than patch out the transforms, is it enough to patch the `get` functions? I feel like that would make it simpler. If not, then I get it